controllers/nodehealthcheck_controller.go

```diff
@@ -88,30 +108,44 @@ type NodeHealthCheckReconciler struct {
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-func (r *NodeHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, err error) {
+func (r *NodeHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, returnErr error) {
 	log := r.Log.WithValues("NodeHealthCheck name", req.Name)
 
-	// fetch nhc
+	// get nhc
 	nhc := &remediationv1alpha1.NodeHealthCheck{}
-	err = r.Get(ctx, req.NamespacedName, nhc)
+	err := r.Get(ctx, req.NamespacedName, nhc)
 	result = ctrl.Result{}
 	if err != nil {
-		log.Error(err, "failed fetching Node Health Check", "object", nhc)
+		log.Error(err, "failed getting Node Health Check", "object", nhc)
 		if apierrors.IsNotFound(err) {
 			return result, nil
 		}
 		return result, err
 	}
 
-	// check if we need to patch status before we exit Reconcile
+	resourceManager := resources.NewManager(r.Client, ctx)
+
+	// always check if we need to patch status before we exit Reconcile
+	// skip inFlightRemediations until we know we have valid templates, else it will fail status update
+	patchInFlightRemediations := false
 	nhcOrig := nhc.DeepCopy()
 	defer func() {
-		err = r.patchStatus(nhc, nhcOrig)
-		if err != nil {
-			log.Error(err, "failed to patch NHC status")
+		patchErr := r.patchStatus(nhc, nhcOrig, patchInFlightRemediations, resourceManager)
+		if patchErr != nil {
+			log.Error(err, "failed to update status")
+			// check if we have an error from the rest of the code already
+			if returnErr != nil {
+				returnErr = utilerrors.NewAggregate([]error{patchErr, returnErr})
+			} else {
+				returnErr = patchErr
+			}
 		}
 	}()
 
+	// set counters to zero for disabled NHC
+	nhc.Status.ObservedNodes = 0
+	nhc.Status.HealthyNodes = 0
```

Comment:
NIT: I believe it is redundant, since these status fields are `int` and there initalized with zero. In addition we also set the [default value to be zero](https://github.com/medik8s/node-healthcheck-operator/blob/main/api/v1alpha1/nodehealthcheck_types.go#L123-L129).

Reply:
correct, but they can have other values at this point. I prefer zero over outdated values, in case we don't reach the new calculation.